db/migrate/20210720065105_change_participant_declarations_to_common_base_class.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class ChangeParticipantDeclarationsToCommonBaseClass < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  def change
+    safety_assured do
+      change_table :participant_declarations, bulk: true do |t|
+        t.string :course_identifier, null: true
+        t.string :evidence_held, null: true
+        t.string :type, default: "ParticipantDeclaration::ECF"
```

Comment:
Warning: this will mean anyone who tries making an abstract `ParticipantDeclaration` will end up with an ECF one, no real way of stopping it. I made that mistake with tokens.

If we don't want that, uhh, how many declarations are there in live system? We might just get away with a migration setting the type on all of them.

Reply:
Don't really like enabling people to make Abstract instances. When the next bit goes in, it's going to have restriction and validations to prevent that. It'll either have to be ECF or NPQ. Base ones won't make sense and will fail unless the correct course types are set.